src/lib.rs

```diff
@@ -0,0 +1,398 @@
+    lib: bool,
+    /// Build only the specified binary
+    #[structopt(long, group = "target_type")]
+    bin: Option<String>,
+    /// Build only the specified example
+    #[structopt(long, group = "target_type")]
+    example: Option<String>,
+    /// Build only the specified test target
+    #[structopt(long, group = "target_type")]
+    test: Option<String>,
+    /// Build only the specified benchmark target
+    #[structopt(long, group = "target_type")]
+    bench: Option<String>,
+}
+
+impl Default for Target {
+    fn default() -> Self {
+        Self {
+            lib: false,
+            bin: None,
+            example: None,
+            test: None,
+            bench: None,
+        }
+    }
+}
+
+impl Target {
+    /// Only build the library portion of the selected package
+    pub fn only_lib(mut self) -> Self {
+        self.lib = true;
+        self.bin.take();
+        self.example.take();
+        self.test.take();
+        self.bench.take();
+
+        self
+    }
+
+    /// Only build the specified binary
+    pub fn only_bin(mut self, s: &str) -> Self {
+        self.lib = false;
+        self.bin = Some(s.to_string());
+        self.example.take();
+        self.test.take();
+        self.bench.take();
+
+        self
+    }
+
+    /// Only build the specified example
+    pub fn only_example(mut self, s: &str) -> Self {
+        self.lib = false;
+        self.bin.take();
+        self.example = Some(s.to_string());
+        self.test.take();
+        self.bench.take();
+
+        self
+    }
+
+    /// Only build the specified test case
+    pub fn only_test(mut self, s: &str) -> Self {
+        self.lib = false;
+        self.bin.take();
+        self.example.take();
+        self.test = Some(s.to_string());
+        self.bench.take();
+
+        self
+    }
+
+    /// Only build the specified benchmark
+    pub fn only_bench(mut self, s: &str) -> Self {
+        self.lib = false;
+        self.bin.take();
+        self.example.take();
+        self.test.take();
+        self.bench = Some(s.to_string());
+
+        self
+    }
+}
+
+/// Specify additional build options
+#[derive(Debug, StructOpt)]
+#[structopt(rename_all = "kebab-case")]
+pub struct BuildExt {
+    /// Selects the stdout output format
+    #[structopt(long, default_value = "human", parse(try_from_str))]
+    message_format: MessageFormat,
+    /// Selects the type of JavaScript runtime which will be generated
+    ///
+    /// Valid only for the `wasm32-unknown-unknown` target.
+    #[structopt(long, parse(try_from_str))]
+    runtime: Option<RuntimeKind>,
+}
+
+impl Default for BuildExt {
+    fn default() -> Self {
+        Self {
+            message_format: MessageFormat::Json,
+            runtime: None,
+        }
+    }
+}
+
+impl BuildExt {
+    /// Set the message format (for progress messages on stdout).
+    /// Acceptable values are `"human"` (default) and `"json"`.
+    pub fn with_message_fmt(mut self, s: &str) -> Result<Self, Error> {
```

Comment:
This should just take the enum instead of the string; we just need to make it non-exhaustive.